### PR TITLE
Add pylontech integration

### PR DESCRIPTION
**Pull request by functionpointer**
## Description:

Adds documentation for the new pylontech integration.

**Related issue (if applicable):** none

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#4688

## Checklist:

  - [X] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [X] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by functionpointer** [comment]
Not stale, parent is just still pending after 2 months with no response

File: components/pylontech.rst [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+      - Pylontech RX
+      - ESPHome TX via transceiver
+    * - 3
+      - Pylontech TX
+      - ESPHome RX via transceiver
+    * - 4
+      - GND
+      - GND
+      
+Configuration
```
Comment (jesserockz) [inline components/pylontech.rst]:
```suggestion
Component/Hub
```

File: components/pylontech.rst [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    uart:
+      - id: pylontech_uart
+        tx_pin: GPIO1
+        rx_pin: GPIO3
+        baud_rate: 115200
+
```
Comment (jesserockz) [inline components/pylontech.rst]: We don't add external examples here. It is up to the user to visit the uart docs to see how to configure that for their use case.
```suggestion
    # Example configuration entry
```

File: components/pylontech.rst [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    pylontech:
+      - uart_id: pylontech_uart
+        id: pylontech0
+        update_interval: 5s
```
Comment (jesserockz) [inline components/pylontech.rst]: Examples only needs the most basic configuration required.
```suggestion
    pylontech:
```

File: components/pylontech.rst [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+------
+
+All values are reported for every Pylontech battery individually.
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    sensor:
+      - platform: pylontech
+        pylontech_id: pylontech0
```
Comment (jesserockz) [inline components/pylontech.rst]: When the example is used with only one component this is not needed
```suggestion
```

File: components/pylontech.rst [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+        
+        voltage:
+          id: bat1_voltage
+          name: "Battery1 Voltage"
+        current:
+          id: bat1_current
+          name: "Battery1 Current"
+        coulomb:
+          id: bat1_soc
+          name: "Battery1 State of Charge"
```
Comment (jesserockz) [inline components/pylontech.rst]:
```suggestion
        voltage:
          name: "Battery1 Voltage"
        current:
          name: "Battery1 Current"
        coulomb:
          name: "Battery1 State of Charge"
```

File: components/pylontech.rst [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+        
+        voltage:
+          id: bat2_voltage
+          name: "Battery2 Voltage"
+        current:
+          id: bat2_current
+          name: "Battery2 Current"
+        coulomb:
+          id: bat2_soc
+          name: "Battery2 State of Charge"
```
Comment (jesserockz) [inline components/pylontech.rst]:
```suggestion
```

File: components/pylontech.rst [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+
+Configuration variables:
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+- **pylontech_id** (**Required**): ID of the pylontech instance
```
Comment (jesserockz) [inline components/pylontech.rst]:
```suggestion
- **pylontech_id** (*Optional*): Manually specify the ID of the pylontech instance if there are multiple.
```

File: components/pylontech.rst [hunk 8]
```diff
@@ -0,0 +1,229 @@
...
+- **battery** (**Required**): Which battery to monitor. 1 stands for the master battery, 2..6 for slave batteries.
```
Comment (jesserockz) [inline components/pylontech.rst]: We try not to use master/slave terms
```suggestion
- **battery** (**Required**): Which battery to monitor. 1 stands for the main battery, 2..6 for child batteries.
```

File: components/pylontech.rst [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+- **voltage** (*Optional*): Voltage. Should be around 48V
```
Comment (jesserockz) [inline components/pylontech.rst]: Also needs to apply to the following sensors and similar for text sensors.
```suggestion
- **voltage** (*Optional*): Voltage of the battery. All options from :ref:`Sensor <config-sensor>`.
```

File: components/pylontech.rst [hunk 10]
```diff
@@ -0,0 +1,229 @@
...
+- **current** (*Optional*): Current
+- **coulomb** (*Optional*): State of Charge in percent
+- **temperature** (*Optional*): Temperature
+- **temperature_low** (*Optional*): Historic minimum temperature
+- **temperature_high** (*Optional*): Historic maximum temperature
+- **voltage_low** (*Optional*): Historic minimum voltage
+- **voltage_high** (*Optional*): Historic maximum voltage
+- **mos_temperature** (*Optional*): Temperature of the mosfets.
+
+All sensors are normal sensors, so each one supports options from :ref:`Sensor <config-sensor>`.
```
Comment (jesserockz) [inline components/pylontech.rst]: Inline is better as it shows up with code completion dialogs
```suggestion
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Review by jesserockz — APPROVED** [review]
Thanks @functionpointer

**Pull request merged**